.changeset/four-turtles-suffer.md

```diff
@@ -0,0 +1,5 @@
+---
+"effect": minor
+---
+
+add key type to ReadonlyRecord
```

Comment:
if we add the K parameter then I would also try to handle symbols as keys (which is something I've wanted to do for a long time)

Reply:
I actually did add that then reverted it.

Will add it back later.